java/client/src/org/openqa/selenium/chrome/ChromeOptions.java

```diff
@@ -89,9 +76,78 @@ public ChromeOptions() {
   @Override
   public ChromeOptions merge(Capabilities extraCapabilities) {
     super.merge(extraCapabilities);
+    if (extraCapabilities instanceof ChromeOptions) {
+      Map options = (Map) extraCapabilities.asMap().get("goog:chromeOptions");
+      mergeArguments(options);
+      mergeExtensions(options);
+      mergeExtensionFiles(extraCapabilities);
+      mergeBinary(options);
+      mergeExperimentalOptions(extraCapabilities);
+    }
     return this;
   }
 
+  private void mergeArguments(Map options) {
+    addArguments((List<String>) options.get("args"));
+    deleteDuplicates((ArrayList) args);
+  }
+
+  private void mergeExtensions(Map options) {
+    if (options.get("extensions") != null) {
+      addEncodedExtensions((List<String>) options.get("extensions"));
+      deleteDuplicates((ArrayList) extensions);
+    }
+  }
+
+  private void mergeExtensionFiles(Capabilities extraCapabilities) {
+    try {
+      Field extensionFilesField = ChromeOptions.class.getDeclaredField("extensionFiles");
+      extensionFilesField.setAccessible(true);
+      List<File> extensionFilesList = (List<File>) extensionFilesField.get(extraCapabilities);
+      for (File path: extensionFilesList) {
+        addExtensions(path);
+      }
+      deleteDuplicates((ArrayList) extensionFiles);
+    } catch (NoSuchFieldException | IllegalAccessException e) {
+      e.printStackTrace();
```

Comment:
Rethrow the exception, even if you're sure it'll never happen. A `WebDriverException` would work here.